tests/foreman/destructive/test_ansible.py

```diff
@@ -63,7 +66,23 @@ def test_positive_import_all_roles(target_sat):
 
     :expectedresults: All roles are imported successfully. One role is deleted successfully.
     """
-    with target_sat.ui_session() as session:
+    password = settings.server.admin_password
+    if auth_type == 'non-admin':
+        ansible_manager_role = target_sat.api.Role().search(
+            query={'search': 'name="Ansible Roles Manager"'}
+        )
+        username = gen_string('alphanumeric')
+        user = target_sat.api.User(
+            role=ansible_manager_role,
+            admin=True,
+            login=username,
+            password=password,
+            organization=[function_org],
+        ).create()
+        request.addfinalizer(user.delete)
+    username = user.login if auth_type == 'non-admin' else settings.server.admin_username
```

Comment:
above if statement can be merged in the previous if statement it self, non-blocking

Reply:
@omkarkhatavkar Good point, updated!